README.md

```diff
@@ -272,8 +274,21 @@ const customer = client.Customer({
     onQueryError({ credentials, query, reportOptions, error }) {
       throw new Error(error);
     },
-    onQueryEnd({ credentials, query, reportOptions, response }) {
-      return response.slice(0, 5);
+    onQueryEnd({ credentials, query, reportOptions, response, resolve }) {
+      resolve(response.slice(0, 5)); // resolves the query with the given argument
+    },
+    onMutationStart({ credentials, mutations, cancel }) {
+      if (mutations.length === 0) {
+        cancel({}); // cancels the mutation and returns the given argument
+      }
+    },
+    onMutationError({ credentials, mutations, error }) {
+      throw new Error(error);
```

Comment:
Very minor but we should change these error hook examples to just log the error instead of throwing it. This isn't the intended use of the error hook, and would definitely cause problems. A log or calling a log service would be a better example.

Reply:
Also, it would be good to have a note that the mutation hooks only currently work with `mutateResources` and not the individual service methods such as `create`, `update` and `delete`.